### DISCUSSION TITLE
keepassXC snap and appimage

### DISCUSSION THREAD

**Discussion by HDInfautre** [keepassxreboot/keepassxc#5841] [Q&A] [answered]
Hello
  By opening keepassXC several bases: personal, work, associations ... keepassXC stores them in the tab or last opened bases; 
Cool it is practical but I would have liked that 2 bases are not memorized. 
I wish I didn't have to erase the history all the time because it is very useful.

My default install is done via the ubuntu 20.04 installation (snap).
On my external disk that I can move to another computer, I installed KeePassXC-2.6.2-x86_64.AppImage. Unfortunately, when I open keepass XC via snap or appimage, I open the application with the same history.

On some appImage applications, it is possible to limit the reading to a directory without interacting with the snap application (firefox).
Is it possible with keepassXC? without development because it is not a developer! roadmap?
Regards

**Comment by droidmonkey** [comment]
I am sorry but I have no idea what you are referring to. Can you add screenshots? Please don't take screenshot of your sensitive data though.

**Comment by HDInfautre** [comment]
HI,

I'm sorry but I don't think a screenshot will help because in both cases I get the same thing. Sorry for my bad english.

I try to rephrase it differently
I have firefox1 (default snap) with several profiles that are all visible in this firefox.
On a disk I have a firefox2 (appimage) with one profile.
=> There is no sharing of profiles between them!

Keepass1 (snap) installed on the computer has several bdd in its history
I want to have 2nd keepass2 (appimage) in a hard disk directory which, once launched, does not recover the other keepass1 databases.
Is it clearer?
Regards

**Comment by HDInfautre** [comment]
Hello
Sorry to bother you but as this may be more difficult for the next 15 days, can you tell me if this is feasible and if so how?
Regards

**Comment by phoerious** [comment]
I still don't know what you are talking about. Please try to depict what you want to achieve with annotated screenshots.

**Comment by HDInfautre** [comment]
Hello
Ok here are the pictures
kdbx open with keepassXC (appimage) - do not show kdbx open with keepassXC (snap) and vice versa
Is this clearer?
the equivalent for firefox:
https://docs.appimage.org/user-guide/portable-mode.html
https://discourse.appimage.org/t/portable-configuration-data/105/10
is it possible to make an equivalent?

![keepassXC_appimage](https://user-images.githubusercontent.com/6817936/102768780-faa6a700-4381-11eb-9325-34eed9bad9ab.png)
![keepassXC_snap](https://user-images.githubusercontent.com/6817936/102768789-fda19780-4381-11eb-863a-8717b4bbe2b5.png)

**Comment by phoerious** [comment]
I believe that depends on what config file the Snap version uses. I suppose it's the same @droidmonkey?
We certainly don't support individual profiles, but you can add a .portable file next to your AppImage. That should switch to "portable" mode and ignore your global keepassxc.ini. We do it for the Windows version, I never tried if it actually works with the AppImage.

**Reply by droidmonkey** [reply]
The snap already uses its own configuration file located in ~/snap/keepassxc/common/.config

It is pretty clear that the user is not actually using two different versions of keepassxc and is likely either always launching the snap version or the appimage version.

**Reply by phoerious** [reply]
That's what I thought. I was rather confused by the Snap supposedly using the same config.

**Comment by HDInfautre** [comment]
Hello
Yes I understand that most users only have one version of keepassXC.
But as with firefox, users may have different needs. And if I understood correctly appimage would allow you to have your own config file to achieve what you want!

I didn't understand if it's possible?
If this is not possible right now? is it possible to consider it in a roadmap ?

**Reply by droidmonkey** [reply]
On each application you run, open Help -> About, click the Debug Info tab and copy paste that info into the chat here. For both applications, appimage and snap.

**Comment by HDInfautre** [comment]
SNAP
```
KeePassXC - Version 2.6.2
Révision : e9b9582

Qt 5.12.8
Le mode débogage est désactivé.

Système d’exploitation : Ubuntu 20.04.1 LTS
Architecture de l’unité centrale : x86_64
Noyau : linux 5.4.0-58-generic

Extensions activées :
- Saisie automatique
- Intégration aux navigateurs
- Agent SSH
- KeeShare (partage signé et non signé)
- YubiKey
- Intégration au Secret Service

Bibliothèques cryptographiques :
 libgcrypt 1.8.5
```


APPIMAGE
```
KeePassXC - Version 2.6.2
Révision : e9b9582
Canal : AppImage

Qt 5.14.2
Le mode débogage est désactivé.

Système d’exploitation : Ubuntu 20.04.1 LTS
Architecture de l’unité centrale : x86_64
Noyau : linux 5.4.0-58-generic

Extensions activées :
- Saisie automatique
- Intégration aux navigateurs
- Agent SSH
- KeeShare (partage signé et non signé)
- YubiKey
- Intégration au Secret Service

Bibliothèques cryptographiques :
 libgcrypt 1.8.1
```

**Reply by droidmonkey** [reply]
You are not running the snap, that is either the PPA version or system version. You would see "Canal: Snap". Since you are not running the snap you are in fact using the same configuration file for both instances. You need to either run the AppImage with "--localconfig ~/.config/keepassxc_appimage/keepassxc.ini" or add a ".portable" file in the same directory as the appimage.

**Comment by HDInfautre** [comment]
Ok sorry, I thought ubuntu had installed it in snap format.
But I'm not sure I understood what you are proposing because keepassXC appimage will be on a USB key or external drive, sometimes on my computer but also elsewhere.
Can I move the config file to my disk/key so that I don't have a problem of rights on another computer?
Is it preferable that I install the snap package instead of the ppa to simplify use?

**Reply by droidmonkey** [reply]
It doesn't matter where you specify the config file. Or you could actually install the snap version and this'll be a non issue.

**Comment by HDInfautre** [comment]
Hi,

I repeat what I understood (I don't speak good English and I'm not a developer)
It is possible to have 2 keepassXC at the same time.
It is not easier with snap + appimage, so I stay in ppa + appimage.

1/ I have created directories (cf image)
![Capture d’écran de 2020-12-22 10-47-29](https://user-images.githubusercontent.com/6817936/102874873-54bb7100-4443-11eb-8d66-01155e997fd7.png)
2/ I copied/pasted the keepassxc.ini I found under home/xxx/.config/keepassxc
3/ I placed the order but I get the message "order not found".
`......KeePassXC-portable$ KeePassXC-2.6.2-x86_64.AppImage --localconfig ~/.config/keepassxc_appimage/keepassxc.ini
KeePassXC-2.6.2-x86_64.AppImage : commande introuvable

`
I did not understand what to do with the option:
"add a "... .portable" file in the same directory as the appimage .."

what do I have to do to finalize?
Regards

**Reply by phoerious** [reply]
You have to place a file with that name next to the AppImage. And the config location is called config, not .config. But that will be created automatically.

**Comment by HDInfautre** [comment]
HI
I am sorry to ask you again but it is very clear to you but not to me.
I have a partial answer to my questions with a short sentence.
I make a lot of unsuccessful attempts at interpretation;
>You have to place a file with that name (keepassxc.ini?) next to the AppImage. 
And the config location is called config (directory I create?, not .config. But that will be created automatically (when? how?)

Can you tell me with my directory structure what I need to do (there shouldn't be many steps if I understand correctly)? and
 what I have to launch on the command line (is that right?)
Regards

**Reply by phoerious** [reply]
```
USB Drive
|
| - KeePassXC.AppImage
| - .portable
```

**Comment by HDInfautre** [comment]
re
Further tests (see picture) without success
![Capture d’écran de 2020-12-22 15-10-29](https://user-images.githubusercontent.com/6817936/102897363-27cd8500-4468-11eb-90d8-b8c2a90093d9.png)

1/ with .portable (file), essai2
I click on the application no file created! I'm looking at the search history, do I need to save a change of settings?

2/ with config (directory), essai3
I launch the following command:
`KeePassXC-portable/essai3$ KeePassXC-2.6.2-x86_64.AppImage --localconfig ~/config/keepassxc_appimage/keepassxc.ini
KeePassXC-2.6.2-x86_64.AppImage : commande introuvable
`

**Reply by phoerious** [reply]
Please just use the Snap.

**Comment by HDInfautre** [comment]
What annoys me with snap (already tried) is that I seem to have had access rights problems because my bases (kdbx) are under /media/ and maybe loss of general configuration!

But above all I would prefer that another person (eventually) corrects the stupid errors of my tests 2 & 3 on the provided image to understand but especially to finalize because I already spent a few hours, including looking for documentation before asking you (suggestion to make a tutorial one day to create 2 instances).
Thank you

**Comment by droidmonkey** [comment, marked as answer]
Please read here, this explains the command line options https://keepassxc.org/docs/KeePassXC_UserGuide.html#_command_line_options

We won't be able to provide additional support since this is on you to figure out your desired running condition.
